subsys/bluetooth/host/buf.c

```diff
@@ -141,3 +141,36 @@ struct net_buf *bt_buf_get_evt(uint8_t evt, bool discardable,
 		return bt_buf_get_rx(BT_BUF_EVT, timeout);
 	}
 }
+
+#ifdef ZTEST_UNITTEST
+#if defined(CONFIG_BT_HCI_ACL_FLOW_CONTROL)
+struct net_buf_pool *bt_buf_get_evt_pool(void)
+{
+	return &evt_pool;
+}
+
+struct net_buf_pool *bt_buf_get_acl_in_pool(void)
+{
+	return &acl_in_pool;
+}
+#else
+struct net_buf_pool *bt_buf_get_hci_rx_pool(void)
+{
+	return &hci_rx_pool;
+}
+#endif
+
+#if defined(CONFIG_BT_BUF_EVT_DISCARDABLE_COUNT)
+struct net_buf_pool *bt_buf_get_discardable_pool(void)
+{
+	return &discardable_pool;
+}
+#endif
+
+#if defined(CONFIG_BT_CONN) || defined(CONFIG_BT_ISO)
+struct net_buf_pool *bt_buf_get_num_complete_pool(void)
+{
+	return &num_complete_pool;
+}
+#endif
+#endif
```

Comment:
Could you make this `#endif /* ZTEST_UNITTEST */` to make it clear which condition it is terminating?  It’s in general a good idea to do that, especially for long blocks and when you have many ifdefs in the same file.